csharp/src/Drivers/Apache/Spark/SparkConnection.cs

```diff
@@ -688,9 +693,65 @@ private string PatternToRegEx(string? pattern)
 
             return builder.ToString();
         }
+
+        /// <summary>
+        /// Provides a parser for SQL DECIMAL type definitions.
+        /// </summary>
+        private static class SqlDecimalTypeParser
+        {
+            // Pattern is based on this definition
+            // https://docs.databricks.com/en/sql/language-manual/data-types/decimal-type.html#syntax
+            // { DECIMAL | DEC | NUMERIC } [ (  p [ , s ] ) ]
+            // p: Optional maximum precision (total number of digits) of the number between 1 and 38. The default is 10.
+            // s: Optional scale of the number between 0 and p. The number of digits to the right of the decimal point. The default is 0.
+            private static readonly Regex s_expression = new(
```

Comment:
Wonder, can this be const instead of static readonly?

Reply:
In C#, only constant expressions can be assigned to a `const` variable. Anything with `new` requires the runtime to run to create the memory behind the variable. So this is the "best" equivalent to `const` and has the same functional behaviour.